common/views/components/Checkbox/Checkbox.js

```diff
@@ -0,0 +1,86 @@
+// @flow
+
+import React from 'react';
+import styled from 'styled-components';
+import { classNames } from '../../../utils/classnames';
+import Space from '../styled/Space';
+
+const CheckboxLabel = styled.label.attrs({
+  className: classNames({
+    'flex flex--v-center': true,
+  }),
+})``;
+
+const CheckboxBox = styled.span.attrs({
+  className: classNames({
+    CheckboxBox: true,
+    'flex-inline flex--v-center flex--h-center relative': true,
+  }),
+})`
+  width: 1em;
+  height: 1em;
+  border: 1px solid ${props => props.theme.colors.silver};
+  border-radius: 3px;
+  transition: all 400ms ease;
+
+  &:after {
+    content: '';
+    position: absolute;
+    width: 60%;
+    height: 60%;
+    opacity: 0;
+    transition: opacity 400ms ease;
+    background: url('data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A//www.w3.org/2000/svg%22%20viewBox%3D%220%200%2032%2032%22%3E%0A%20%20%3Cpath%20fill%3D%22%23ffffff%22%20d%3D%22M6%2012l-6%206%2012%2012%2020-20-6-6-14%2014z%22%3E%3C/path%3E%0A%3C/svg%3E%0A');
+  }
+`;
+
+const CheckboxInput = styled.input.attrs({
+  type: 'checkbox',
+})`
+  position: absolute;
+  opacity: 0;
+  z-index: 1;
+  width: 1em;
+  height: 1em;
+
+  &:checked ~ .CheckboxBox {
+    background: ${props => props.theme.colors.green};
+    border-color: ${props => props.theme.colors.green};
+
+    &:after {
+      opacity: 1;
+    }
+  }
+`;
+
+type CheckboxProps = {|
+  id: string,
+|};
+type CheckboxWithLabelProps = {|
+  ...CheckboxProps,
+  text: string,
+|};
+
+export function Checkbox({ id, ...inputProps }: CheckboxProps) {
+  return (
+    <>
+      <CheckboxInput id={id} {...inputProps} />
+      <CheckboxBox />
+    </>
+  );
+}
+
+function CheckboxWithLabel({
+  id,
+  text,
+  ...inputProps
+}: CheckboxWithLabelProps) {
+  return (
+    <CheckboxLabel htmlFor={id}>
+      <Checkbox id={id} {...inputProps} />
+      <Space h={{ size: 'xs', properties: ['margin-left'] }}>{text}</Space>
+    </CheckboxLabel>
+  );
+}
+
+export default CheckboxWithLabel;
```

Comment:
Why not just call this `CheckBoxWithLabel`?

Reply:
Do you mean call the file `CheckboxWithLabel`?

Reply:
I think given that, like you say, a checkbox _should_ have a label to be accessible, the component name makes sense as just, `Checkbox` – I'll update the inner render function to be `CheckboxWithoutLabel`.

Reply:
Good call.